marlowe-actus/test/Spec/Marlowe/Util.hs

```diff
@@ -0,0 +1,259 @@
+     , ct_PPEF          = readMaybe (maybeConcatPrefix "PPEF_" (Map.lookup "prepaymentEffect" terms')) :: Maybe PPEF
+     , ct_cPYRT         = 0.0
+     , ct_RRCL          = parseMaybeCycle $ Map.lookup "cycleOfRateReset" terms'
+     , ct_RRANX         = parseMaybeDate $ Map.lookup "cycleAnchorDateOfRateReset" terms'
+     , ct_RRNXT         = readMaybe $ Map.lookup "nextResetRate" terms' :: Maybe Double
+     , ct_RRSP          = readMaybe $ Map.lookup "rateSpread" terms' :: Maybe Double
+     , ct_RRMLT         = readMaybe $ Map.lookup "rateMultiplier" terms' :: Maybe Double
+     , ct_RRPF          = readMaybe $ Map.lookup "periodFloor" terms' :: Maybe Double
+     , ct_RRPC          = readMaybe $ Map.lookup "periodCap" terms' :: Maybe Double
+     , ct_RRLC          = readMaybe $ Map.lookup "lifeCap" terms' :: Maybe Double
+     , ct_RRLF          = readMaybe $ Map.lookup "lifeFloor" terms' :: Maybe Double
+     , ct_RRMO          = Map.lookup "marketObjectCodeOfRateReset" terms'
+     , enableSettlement          = False
+     , constraints      = Nothing
+     , collateralAmount = 0
+     }
+
+readMaybe :: (Read a) => Maybe String -> Maybe a
+readMaybe term =
+  case term of
+    Just t ->
+      Just $ read t
+    Nothing ->
+      Nothing
+
+parseMaybeDate :: Maybe String -> Maybe Day
+parseMaybeDate date =
+  case date of
+    Just d ->
+      parseDate d
+    Nothing ->
+      Nothing
+
+parseDate :: String -> Maybe Day
+parseDate date =
+  let format | List.length date == 19 = "%Y-%-m-%-dT%T"
+             | otherwise = "%Y-%-m-%-dT%H:%M"
+  in
+    parseTimeM True defaultTimeLocale format date :: Maybe Day
+
+parseMaybeCycle :: Maybe String -> Maybe Cycle
+parseMaybeCycle stringCycle =
+  case stringCycle of
+    Just (_:stringCycle') ->
+      let n = read (takeWhile (< 'A') stringCycle') :: Integer
+          [p, _, s] = dropWhile (< 'A') stringCycle'
+      in
+          Just Cycle { n = n, p = read $ "P_" ++ [p] :: Period, stub = parseStub [s], includeEndDay = False }
+    Nothing ->
+      Nothing
+
+
+parseStub :: String -> Stub
+parseStub stub =
+  case stub of
+    "0" ->
+      LongStub
+    "1" ->
+      ShortStub
+
+maybeDCCFromString :: Maybe String -> Maybe DCC
+maybeDCCFromString stringDCC =
+  case stringDCC of
+    Just dcc ->
+      case dcc of
+        "AA" ->
+          Just DCC_A_AISDA
+        "A360" ->
+          Just DCC_A_360
+        "A365" ->
+          Just DCC_A_365
+        "30E360" ->
+          Just DCC_E30_360
+        _ ->
+          Nothing
+    Nothing ->
+      Nothing
+
+maybeConcatPrefix :: String -> Maybe String -> Maybe String
+maybeConcatPrefix prefix term =
+  case term of
+    Just t ->
+      Just $ prefix ++ t
+    Nothing ->
+      Nothing
+
+maybeReplace :: String -> String -> Maybe String -> Maybe String
+maybeReplace from to string =
+  case string of
+    Just s ->
+      Just $ replace from to s
+    Nothing ->
+      Nothing
```

Comment:
If you want to act only on the `Just` of a `Maybe` you can use `fmap` like you would use `map` in a list with 0/1 elements.
For example this function would become:
```haskell
maybeReplace :: String -> String -> Maybe String -> Maybe String
maybeReplace from to string = fmap (replace from to) string
```